Add cascade parameter, update Uploader version

begoldsm opened the pull request:
Update ADLS upload/download package to use latest released version.
Explicitly add the ability to specify the cascade parameter (which
defaulted to false when not present) to allow for deletion of
credentials and all things that depend on them (similar to a recursive
delete).

Commit:
Add cascade parameter, update Uploader version

Update ADLS upload/download package to use latest released version.
Explicitly add the ability to specify the cascade parameter (which
defaulted to false when not present) to allow for deletion of
credentials and all things that depend on them (similar to a recursive
delete).

Commit:
Fix credential test

Now that cascade is explicitly in each call, needed to rerun the tests
to record with it in the header.

begoldsm commented:
@markcowl @cormacpayne could you guys take a quick look at this PR. It is a version bump + improvement for catalog and the upload logic.

markcowl commented on src/ResourceManagement/DataLake.Analytics/Microsoft.Azure.Management.DataLake.Analytics/Generated/CatalogOperations.cs:
```diff
@@ -1643,7 +1648,7 @@ internal CatalogOperations(DataLakeAnalyticsCatalogManagementClient client)
         /// A response object containing the response body and response headers.
         /// </return>
-        public async Task<AzureOperationResponse> DeleteCredentialWithHttpMessagesAsync(string accountName, string databaseName, string credentialName, DataLakeAnalyticsCatalogCredentialDeleteParameters parameters = default(DataLakeAnalyticsCatalogCredentialDeleteParameters), Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = default(CancellationToken))
+        public async Task<AzureOperationResponse> DeleteCredentialWithHttpMessagesAsync(string accountName, string databaseName, string credentialName, DataLakeAnalyticsCatalogCredentialDeleteParameters parameters = default(DataLakeAnalyticsCatalogCredentialDeleteParameters), bool? cascade = false, Dictionary<string, List<string>> customHeaders = null, CancellationToken cancellationToken = defaul ...
```
Test coverage for the new API parameter?

begoldsm replied:
Good call. I will add it now.

Commit:
Address PR comments and revert temp change.

I had an experimental change in the uploader to explicitly set the sync
flag. This did not improve performance and would have created extra
complexity in the code, so it should have been fully reverted. I missed
one file and have now reverted that.
